src/lambda_function/s3_to_glue/app.py

```diff
@@ -31,38 +73,56 @@ def submit_s3_to_json_workflow(messages: list[dict[str,str]], workflow_name):
-        RunProperties={"messages": json.dumps(messages)},
+        RunProperties={"messages": json.dumps(objects_info)},
     )
 
 
-def lambda_handler(event, context):
-    """
-    The Lambda entrypoint
-
-    Given a list of files in s3 bucket to process, submit the keys and bucket
-    of the new files to a S3 to JSON Glue workflow.
+def lambda_handler(event, context) -> None:
+    """This main lambda function will be triggered by a
+        SQS event and will poll the SQS queue for
+        all available S3 event messages
 
     Args:
-        event (dict): A fileview object
-        context (obj) : An object that provides methods and properties that
-        provides info about the invocation, function, and execution environment
-
-    Returns:
-        (None) Submits new file records to a Glue workflow.
+        event (json): SQS event
+        context: NA
     """
-    messages = []
-    for record in event["Records"]:
-        bucket_name = record["s3"]["bucket"]["name"]
-        object_key = record["s3"]["object"]["key"]
-        message = {"source_bucket": bucket_name, "source_key": object_key}
-        if "owner.txt" not in object_key:
-            messages.append(message)
-    if len(messages) > 0:
-        logger.info(
+    # Initialize SQS client
+    sqs = boto3.client("sqs")
+    s3_objects_info = []
+    if "Records" in event:
+        for record in event["Records"]:
+            s3_event_records = json.loads(record["body"])
+            if "Records" in s3_event_records:
+                for s3_event in s3_event_records["Records"]:
+                    bucket_name = s3_event["s3"]["bucket"]["name"]
+                    object_key = s3_event["s3"]["object"]["key"]
+                    object_info = {
+                        "source_bucket": bucket_name,
+                        "source_key": object_key,
+                    }
+                    if filter_object_info(object_info) is not None:
+                        s3_objects_info.append(object_info)
+                    else:
+                        logger.info(
+                            f"Object doesn't meet the S3 event rules to be processed. Skipping\n{object_info}"
+                        )
+
+        if len(s3_objects_info) > 0:
+            logger.info(
                 "Submitting the following files to "
-                f"{os.environ['PRIMARY_WORKFLOW_NAME']}: {json.dumps(messages)}"
-        )
-        submit_s3_to_json_workflow(
-                messages = messages,
-                workflow_name = os.environ["PRIMARY_WORKFLOW_NAME"]
-        )
+                f"{os.environ['PRIMARY_WORKFLOW_NAME']}: {json.dumps(s3_objects_info)}"
+            )
+            submit_s3_to_json_workflow(
+                objects_info=s3_objects_info,
+                workflow_name=os.environ["PRIMARY_WORKFLOW_NAME"],
+            )
+        # Delete the messages from the queue after all events are successfully submitted
+        for record in event["Records"]:
```

Comment:
What is your rationale for doing this loop at the end and not deleting events as we go?

Reply:
Because we call `submit_s3_to_json_workflow` after we get through all of the messages. I don't think we should delete any events until we reach the end where we decide which ones to submit to our S3 to JSON workflow otherwise we would lost that event information if we deleted it earlier and they never make it to the submit workflow step because there was an error with another event.

Reply:
Can we add the rationale as a comment.  It is helpful to tell the readers what the code is doing, but we should be able to do that by reading the code.

Reply:
So lambda automatically deletes messages after processing them so we don't need these lines of code anymore.